From 85c987757fe9a259a370c79333b3e57ef820dcf5 Mon Sep 17 00:00:00 2001
From: AffectedArc07 <25063394+AffectedArc07@users.noreply.github.com>
Date: Tue, 24 Oct 2023 21:44:45 +0100
Subject: [PATCH] Paradise Version Changes


diff --git a/Cargo.lock b/Cargo.lock
index afdaa60..b276d34 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2166,7 +2166,7 @@ dependencies = [
 
 [[package]]
 name = "rust-g"
-version = "3.0.0"
+version = "3.0.0-P"
 dependencies = [
  "aho-corasick",
  "base64",
diff --git a/Cargo.toml b/Cargo.toml
index 958fbd4..e3965dd 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,15 +1,15 @@
 [package]
 name = "rust-g"
 edition = "2021"
-version = "3.0.0"
+version = "3.0.0-P"
 authors = [
     "Bjorn Neergaard <bjorn@neersighted.com>",
     "Tad Hardesty <tad@platymuus.com>",
     "rust-g maintainer team",
 ]
-repository = "https://github.com/tgstation/rust-g"
+repository = "https://github.com/ParadiseSS13/rust-g"
 license = "MIT"
-description = "Offloaded task library for the /tg/ Space Station 13 codebase"
+description = "Offloaded task library for the ParadiseSS13 Space Station 13 codebase - Adapted from tgstation/rust-g"
 
 [lib]
 crate-type = ["cdylib"]
-- 
2.41.0.windows.3

