From ecce1db2d967a128cc02baa73245d84009d9c7ec Mon Sep 17 00:00:00 2001
From: AffectedArc07 <25063394+AffectedArc07@users.noreply.github.com>
Date: Wed, 13 Oct 2021 22:04:37 +0100
Subject: [PATCH] Paradise Logging Changes


diff --git a/dmsrc/log.dm b/dmsrc/log.dm
index d0380a5..f3be525 100644
--- a/dmsrc/log.dm
+++ b/dmsrc/log.dm
@@ -1,2 +1,2 @@
-#define rustg_log_write(fname, text, format) call(RUST_G, "log_write")(fname, text, format)
+#define rustg_log_write(fname, text) call(RUST_G, "log_write")(fname, text)
 /proc/rustg_log_close_all() return call(RUST_G, "log_close_all")()
diff --git a/src/log.rs b/src/log.rs
index 9bd019b..6722353 100644
--- a/src/log.rs
+++ b/src/log.rs
@@ -14,7 +14,7 @@ thread_local! {
     static FILE_MAP: RefCell<HashMap<OsString, File>> = RefCell::new(HashMap::new());
 }
 
-byond_fn!(fn log_write(path, data, ...rest) {
+byond_fn!(fn log_write(path, data) {
     FILE_MAP.with(|cell| -> Result<()> {
         // open file
         let mut map = cell.borrow_mut();
@@ -24,20 +24,10 @@ byond_fn!(fn log_write(path, data, ...rest) {
             Entry::Vacant(elem) => elem.insert(open(path)?),
         };
 
-        if rest.first().map(|x| &**x) == Some("false") {
-            // Write the data to the file with no accoutrements.
-            write!(file, "{}", data)?;
-        } else {
-            // write first line, timestamped
-            let mut iter = data.split('\n');
-            if let Some(line) = iter.next() {
-                write!(file, "[{}] {}\n", Utc::now().format("%F %T%.3f"), line)?;
-            }
-
-            // write remaining lines
-            for line in iter {
-                write!(file, " - {}\n", line)?;
-            }
+        // write all lines timestamped
+        let iter = data.split('\n');
+        for line in iter {
+            write!(file, "[{}] {}\n", Utc::now().format("%FT%T"), line)?;
         }
 
         Ok(())
-- 
2.17.1

