From 06758a8bc4e57d42c354740437a125169a79d759 Mon Sep 17 00:00:00 2001
From: AffectedArc07 <25063394+AffectedArc07@users.noreply.github.com>
Date: Thu, 14 Oct 2021 15:50:14 +0100
Subject: [PATCH] Paradise HTTP Module Tweaks


diff --git a/dmsrc/http.dm b/dmsrc/http.dm
index eba7d68..c880334 100644
--- a/dmsrc/http.dm
+++ b/dmsrc/http.dm
@@ -9,3 +9,5 @@
 #define rustg_http_request_blocking(method, url, body, headers, options) RUSTG_CALL(RUST_G, "http_request_blocking")(method, url, body, headers, options)
 #define rustg_http_request_async(method, url, body, headers, options) RUSTG_CALL(RUST_G, "http_request_async")(method, url, body, headers, options)
 #define rustg_http_check_request(req_id) RUSTG_CALL(RUST_G, "http_check_request")(req_id)
+/proc/rustg_create_async_http_client() return RUSTG_CALL(RUST_G, "start_http_client")()
+/proc/rustg_close_async_http_client() return RUSTG_CALL(RUST_G, "shutdown_http_client")()
diff --git a/src/http.rs b/src/http.rs
index c077f06..ef98d45 100644
--- a/src/http.rs
+++ b/src/http.rs
@@ -1,8 +1,8 @@
 use crate::{error::Result, jobs};
-use once_cell::sync::Lazy;
 use serde::{Deserialize, Serialize};
 use std::collections::{BTreeMap, HashMap};
 use std::io::Write;
+use std::cell::RefCell;
 
 // ----------------------------------------------------------------------------
 // Interface
@@ -63,7 +63,9 @@ byond_fn!(fn http_check_request(id) {
 const VERSION: &str = env!("CARGO_PKG_VERSION");
 const PKG_NAME: &str = env!("CARGO_PKG_NAME");
 
-pub static HTTP_CLIENT: Lazy<ureq::Agent> = Lazy::new(ureq::agent);
+thread_local! {
+    pub static HTTP_CLIENT: RefCell<Option<ureq::Agent>> = RefCell::new(Some(ureq::agent()));
+}
 
 // ----------------------------------------------------------------------------
 // Request construction and execution
@@ -81,38 +83,57 @@ fn construct_request(
     headers: &str,
     options: &str,
 ) -> Result<RequestPrep> {
-    let mut req = match method {
-        "post" => HTTP_CLIENT.post(url),
-        "put" => HTTP_CLIENT.put(url),
-        "patch" => HTTP_CLIENT.patch(url),
-        "delete" => HTTP_CLIENT.delete(url),
-        "head" => HTTP_CLIENT.head(url),
-        _ => HTTP_CLIENT.get(url),
-    }
-    .set("User-Agent", &format!("{PKG_NAME}/{VERSION}"));
-
-    let mut final_body = body.as_bytes().to_vec();
-
-    if !headers.is_empty() {
-        let headers: BTreeMap<&str, &str> = serde_json::from_str(headers)?;
-        for (key, value) in headers {
-            req = req.set(key, value);
-        }
-    }
-
-    let mut output_filename = None;
-    if !options.is_empty() {
-        let options: RequestOptions = serde_json::from_str(options)?;
-        output_filename = options.output_filename;
-        if let Some(fname) = options.body_filename {
-            final_body = std::fs::read(fname)?;
+    HTTP_CLIENT.with(|cell| {
+        let borrow = cell.borrow_mut();
+        match &*borrow {
+            Some(client) => {
+                let mut req = match method {
+                    "post" => client.post(url),
+                    "put" => client.put(url),
+                    "patch" => client.patch(url),
+                    "delete" => client.delete(url),
+                    "head" => client.head(url),
+                    _ => client.get(url),
+                }
+                .set("User-Agent", &format!("{PKG_NAME}/{VERSION}"));
+
+                let mut final_body = body.as_bytes().to_vec();
+
+                if !headers.is_empty() {
+                    let headers: BTreeMap<&str, &str> = serde_json::from_str(headers)?;
+                    for (key, value) in headers {
+                        req = req.set(key, value);
+                    }
+                }
+
+                let mut output_filename = None;
+                if !options.is_empty() {
+                    let options: RequestOptions = serde_json::from_str(options)?;
+                    output_filename = options.output_filename;
+                    if let Some(fname) = options.body_filename {
+                        final_body = std::fs::read(fname)?;
+                    }
+                }
+
+                Ok(RequestPrep {
+                    req,
+                    output_filename,
+                    body: final_body,
+                })
+            }
+
+            // If we got here we royally fucked up
+            None => {
+                let client = ureq::agent();
+                let req = client.get("");
+                let output_filename = None;
+                Ok(RequestPrep {
+                    req,
+                    output_filename,
+                    body: Vec::new(),
+                })
+            }
         }
-    }
-
-    Ok(RequestPrep {
-        req,
-        output_filename,
-        body: final_body,
     })
 }
 
@@ -145,3 +166,17 @@ fn submit_request(prep: RequestPrep) -> Result<String> {
 
     Ok(serde_json::to_string(&resp)?)
 }
+
+byond_fn!(fn start_http_client() {
+    HTTP_CLIENT.with(|cell| {
+        cell.replace(Some(ureq::agent()))
+    });
+    Some("")
+});
+
+byond_fn!(fn shutdown_http_client() {
+    HTTP_CLIENT.with(|cell| {
+        cell.replace(None)
+    });
+    Some("")
+});
diff --git a/src/unzip.rs b/src/unzip.rs
index b0a471a..c43dbbd 100644
--- a/src/unzip.rs
+++ b/src/unzip.rs
@@ -10,9 +10,13 @@ struct UnzipPrep {
 }
 
 fn construct_unzip(url: &str, unzip_directory: &str) -> UnzipPrep {
-    let req = HTTP_CLIENT.get(url);
-    let dir_copy = unzip_directory.to_string();
+    let mut reqh = None;
+    HTTP_CLIENT.with(|client| {
+        reqh = Some(client.borrow_mut().as_ref().unwrap().get(url));
+    });
 
+    let dir_copy = unzip_directory.to_string();
+    let req = reqh.unwrap();
     UnzipPrep {
         req,
         unzip_directory: dir_copy,
-- 
2.41.0.windows.3

