From 9ab3dbcef5ac5033bc7039f8f17d7416a44be715 Mon Sep 17 00:00:00 2001
From: AffectedArc07 <25063394+AffectedArc07@users.noreply.github.com>
Date: Sun, 30 Oct 2022 15:29:37 +0000
Subject: [PATCH] Paradise Remove Debug


diff --git a/Cargo.toml b/Cargo.toml
index 53c35c0..098db21 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,7 +19,6 @@ crate-type = ["cdylib"]
 opt-level = 3
 codegen-units = 1
 lto = true
-debug = true
 
 [dependencies]
 thiserror = "2.0"
-- 
2.41.0.windows.3

