From 3726163f09d0aa5a480185ceb10b56e30087ce1d Mon Sep 17 00:00:00 2001
From: AffectedArc07 <25063394+AffectedArc07@users.noreply.github.com>
Date: Sun, 30 Oct 2022 15:29:37 +0000
Subject: [PATCH] Paradise Remove Debug


diff --git a/Cargo.toml b/Cargo.toml
index b2166a7..1ad5412 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -18,7 +18,6 @@ crate-type = ["cdylib"]
 opt-level = 3
 codegen-units = 1
 lto = true
-debug = true
 
 [dependencies]
 thiserror = "1.0"
-- 
2.35.2.windows.1

