From 0a8cfc9199fc6c8e62a0a9c04b53c51c514c022f Mon Sep 17 00:00:00 2001
From: AffectedArc07 <25063394+AffectedArc07@users.noreply.github.com>
Date: Sun, 2 Jan 2022 23:06:29 +0000
Subject: [PATCH] Paradise Fixes 1


diff --git a/dmsrc/redis_pubsub.dm b/dmsrc/redis-pubsub.dm
similarity index 100%
rename from dmsrc/redis_pubsub.dm
rename to dmsrc/redis-pubsub.dm
diff --git a/src/http.rs b/src/http.rs
index b02b580..eecec99 100644
--- a/src/http.rs
+++ b/src/http.rs
@@ -179,16 +179,16 @@ fn submit_request(prep: RequestPrep) -> Result<String> {
     Ok(serde_json::to_string(&resp)?)
 }
 
-byond_fn! { start_http_client() {
+byond_fn!(fn start_http_client() {
     HTTP_CLIENT.with(|cell| {
         cell.replace(Some(setup_http_client()))
     });
     Some("")
-} }
+});
 
-byond_fn! { shutdown_http_client() {
+byond_fn!(fn shutdown_http_client() {
     HTTP_CLIENT.with(|cell| {
         cell.replace(None)
     });
     Some("")
-} }
+});
-- 
2.33.0.windows.2

